### PR TITLE
SONAR-10134 Add organization parameter in conditions ws

### PR DESCRIPTION
**Pull request by julienlancelot**
…dition

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java [hunk 1]
```diff
@@ -57,10 +59,18 @@ public QualityGateFinder(DbClient dbClient) {
     }
   }
 
+  /**
+   * @deprecated Use {@link #getByOrganizationAndId(DbSession, OrganizationDto, long)}
+   */
+  @Deprecated
   public QualityGateDto getById(DbSession dbSession, long qualityGateId) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/QualityGatesWsSupport.java [hunk 2]
```diff
@@ -92,12 +93,22 @@ OrganizationDto getOrganization(DbSession dbSession, Request request) {
     return checkFoundWithOptional(organizationDto, "No organization with key '%s'", organizationKey);
   }
 
+  /**
+   * @deprecated use {@link #checkCanEdit(QGateWithOrgDto)} instead
+   */
+  @Deprecated
   void checkCanEdit(QualityGateDto qualityGate) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/QualityGatesWsSupport.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [QualityGateFinder.java#L52](https://github.com/SonarSource/sonarqube/blob/2c3e5c0933249a434519f9a12ba8506b38d255bc/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java#L52): Remove this use of "getById"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [QualityGateFinder.java#L52](https://github.com/SonarSource/sonarqube/blob/e5c00d96cd1b4bdccafd7a9583ca8874a709b885/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java#L52): Remove this use of "getById"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/QualityGatesWsSupport.java [hunk 3]
```diff
@@ -92,12 +93,22 @@ OrganizationDto getOrganization(DbSession dbSession, Request request) {
...
     checkNotBuiltInt(qualityGate);
     userSession.checkPermission(ADMINISTER_QUALITY_GATES, defaultOrganizationProvider.get().getUuid());
   }
 
+  void checkCanEdit(QGateWithOrgDto qualityGate) {
+    checkNotBuiltInt(qualityGate);
```
Comment (ehartmann) [inline server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/QualityGatesWsSupport.java]: Typo : I think it should be `checkNotBuiltIn( )` (without the last t)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [QualityGateFinder.java#L52](https://github.com/SonarSource/sonarqube/blob/c0d8065de0dc611f1ac47c0742025b1a1cc867cf/server/sonar-server/src/main/java/org/sonar/server/qualitygate/QualityGateFinder.java#L52): Remove this use of "getById"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**